cvmfs/talk.cc

```diff
@@ -308,10 +311,56 @@ static void *MainTalk(void *data __attribute__((unused))) {
           if (proxies == "") {
               Answer(con_fd, "Failed, no valid proxies\n");
           } else {
-            cvmfs::download_manager_->SetProxyChain(proxies);
+            // get existing fallback proxies
+            vector< vector<download::DownloadManager::ProxyInfo> > proxy_chain;
+            unsigned fallback_group;
+            cvmfs::download_manager_->GetProxyInfo(&proxy_chain, NULL, &fallback_group);
+            std::string fallback_proxies;
+            for (unsigned i = fallback_group; i < proxy_chain.size(); ++i) {
+              if (fallback_proxies.size() != 0)
+                fallback_proxies.append(";");
+              if (proxy_chain[i].size() > 0)
+                // fallback proxies may have only one per group
+                fallback_proxies.append(proxy_chain[i][0].Print());
```

Comment:
This will result in a different fallback proxy configuration if one of the fallback proxies got converted into a load-balance group because it uses round-robin DNS.